### ISSUE TITLE
First login never works

### ISSUE THREAD

**Issue by myier** [OtterBrowser/otter-browser#1108]
Logging in on https://mobile.twitter.com/session/new always fails with a bad credential error the first time. It works the second time. Something must be wrong with cookies or sessions, I don't really know, but it works with Opera 12.

In privacy settings, I have do no track activated, the main cookies switch enabled, always accept cookies, keep until current session is closed and never accept third-party. Referrer is set to Opera 12.16. I'm using weekly builds, the problem has been there for the past four weeks at least.

**Comment by myier** [comment]
I cannot log in on office365 as well, it never stops reloading the page after form submission. outlook.com/owa/office365

**Comment by myier** [comment]
And in fact I think there is a cookie problem, because I was logged in on a squirrelmail website, and after submitting the comment here, my session has expired and I cannot log in again.

**Comment by Emdek** [comment]
@myier, which version are you using?
Could you please attach output of diagnostic report (not directly, just use gist or some other paste service)?

**Comment by myier** [comment]
I'm using latest git versions since two months at least (from [AUR](https://aur.archlinux.org/packages/otter-browser-git)).
Here's the [diagnostic](https://gist.github.com/myier/538bbe58f446c0540895f2b29c4814be). Thanks

**Comment by PStokman** [comment]
Myier, also try with 3rd party cookies enabled. At work we host a cloud storage solution and it's authentication fails when I have 3rd party disabled, even though it should not. On Opera and other browsers, having 3rd party cookies disabled does not matter. My experience may be related, so if you see a change, then it's a problem in handling 3rd party cookies and handling/recognising them properly.

**Comment by myier** [comment]
Indeed, allowing third party cookies makes login work the first time.
Also, I have disabled the 'Remember password' stuff, and when I log in, it still asks if I want to save it, until the next page is loaded.
I see two issues then. Thanks PStokman!

**Comment by Emdek** [comment]
@myier, I'm puzzled, so third party cookies were disabled or not?

**Comment by myier** [comment]
I always disable third party cookies, and apparently that's what causes the bug I first reported. I tried what PStokman suggested, re-enabling them for the test, and doing so makes the issue disappear.

I also always disable the password remembering feature, but I still see the bar appearing at the top of tabs asking if we I want to save it or not, maybe another bug should be opened for that but that's not disrupting normal usage.

**Comment by myier** [comment]
Ok, I don't have the issue with twitter anymore in version 142.
But I still cannot use office365, it keeps reloading two pages alternatively.
Maybe there are some websites that are designed to test cookies that we could try?

**Comment by myier** [comment]
Finally, in beta11, the issue with twitter is back. I have to reload the page or fail the first login to be able to login. Office365 still not working.
I found this cookie testing website and it's not working with Otter: http://www.html-kit.com/tools/cookietester/ . Setting the cookie works, removing it doesn't. I tried it with Firefox and both work. Maybe it's related to this issue.

My configuration is to always accept cookies, and always reject third-party cookies, until browser exits.
